runtimes/protocol/lsp.ts

```diff
@@ -40,6 +41,47 @@ export class AutoParameterStructuresProtocolRequestType<P, R, PR, E, RO>
     }
 }
 
+/**
+ * Extra Client information passed during initialization handshake from Client to Server.
+ */
+export interface AWSClientInitializationOptions {
+    /**
+     * Custom client branding information.
+     */
+    product?: {
+        /**
+         * Product name, may contain spaces.
+         */
+        name: string
+        version: string
+    }
+    /**
+     * Client host platform information.
+     */
+    platform?: {
+        /**
+         * Platform name, may contain spaces.
+         */
+        name: string
+        version: string
+    }
+    /**
+     * Custom Client ID value, set based on client extension internal business logic.
+     */
+    clientId?: string
```

Comment:
What's the difference between `client` and `product`?

Reply:
technically no difference, both point to Client extension integrating with Language Server. `clientId` here is unique identifier set by extension to identify unique installations or sessions of extension. Probable it does not have to be a separate field, and can be nested under `product` object.

Reply:
I have updated the naming in latest commit https://github.com/aws/language-server-runtimes/pull/193/commits/969bad7d3819d50417800ed3eb0298261066b229. Also splitting platform and client info in 2 fields.